examples/move/big_vector/sources/big_vector.move

```diff
@@ -0,0 +1,1274 @@
+
+        /// ID of the tree's root structure. Value of `NO_SLICE` means
+        /// there's no root.
+        root_id: u64,
+
+        /// The last node ID that was allocated.
+        last_id: u64,
+    }
+
+    /// A node in the B+ tree.
+    ///
+    /// If representing a leaf node, there are as many keys as values
+    /// (such that `keys[i]` is the key corresponding to `vals[i]`).
+    ///
+    /// A `Slice<u64>` can also represent an interior node, in which
+    /// case `vals` contain the IDs of its children and `keys`
+    /// represent the partitions between children. There will be one
+    /// fewer key than value in this configuration.
+    public struct Slice<E: store> has store, drop {
+        /// Previous node in the intrusive doubly-linked list data
+        /// structure.
+        prev: u64,
+
+        /// Next node in the intrusive doubly-linked list data
+        /// structure.
+        next: u64,
+
+        keys: vector<u128>,
+        vals: vector<E>,
+    }
+
+    /// Wrapper type around indices for slices. The internal index is
+    /// the ID of the dynamic field containing the slice.
+    public struct SliceRef has copy, drop, store { ix: u64 }
+
+    // === Error Codes ===
+
+    /// Max Slice Size provided is too small.
+    const ESliceTooSmall: u64 = 0;
+
+    /// Max Slice Size provided is too big.
+    const ESliceTooBig: u64 = 1;
+
+    /// Max Fan-out provided is too small.
+    const EFanOutTooSmall: u64 = 2;
+
+    /// Max Fan-out provided is too big.
+    const EFanOutTooBig: u64 = 3;
+
+    /// `BigVector` is not empty.
+    const ENotEmpty: u64 = 4;
+
+    /// Key not found in `BigVector`.
+    const ENotFound: u64 = 5;
+
+    /// Key already exists in `BigVector`.
+    const EExists: u64 = 6;
+
+    /// Found a node in an unexpected state during removal (namely, we
+    /// tried to remove from a node's child and found that it had
+    /// become empty, which should not be possible).
+    const EBadRemove: u64 = 7;
+
+    /// Found a pair of nodes that are expected to be adjacent but
+    /// whose linked list pointers don't match up.
+    const ENotAdjacent: u64 = 8;
+
+    /// Tried to redistribute between two nodes, but the operation
+    /// would have had no effect.
+    const EBadRedistribution: u64 = 9;
+
+    // === Constants ===
+
+    /// Sentinel representing the absence of a slice.
+    const NO_SLICE: u64 = 0;
+
+    /// We will accommodate at least this much fan out before
+    /// splitting interior nodes, so that after the split, we don't
+    /// get an interior node that contains only one child.
+    const MIN_FAN_OUT: u64 = 4;
+
+    /// Internal nodes of `BigVector` can't have more children than
+    /// this, to avoid hitting object size limits.
+    const MAX_FAN_OUT: u64 = 4096;
+
+    /// Leaf nodes of `BigVector` can't be bigger than this, to avoid
+    /// hitting object size limits.
+    const MAX_SLICE_SIZE: u64 = 256 * 1024;
+
+    // === Removal fix-up strategies ===
```

Comment:
These constants could be an `enum` (although ideally they would be a private `enum`).